### DISCUSSION TITLE
Interrupt endpoint and vendor peripheral

### DISCUSSION THREAD

**Discussion by morsisko** [hathach/tinyusb#2160] [Q&A]
Hello,

I would like to ask whether it is possible to use this library to implement interrupt transfer endpoints in vendor peripheral class. If so, is there any example how I would do it? As far as I see there is one example of vendor class device in this repository, but it only shows the usage of control and bulk transfers, so I'm not sure if this is even possible to add interrupt endpoint to the vendor device.

**Comment by hathach** [comment]
yeah, currently vendor driver is only looking for bulk endpoint. Adding support for interrupt isn't too difficult, I will update it later. Meanwhile, you can try to do it with HID Generic first.
